### DISCUSSION TITLE
Error 500 when opening "Settings>General" and tests are failing

### DISCUSSION THREAD

**Discussion by stephdeville** [alexjustesen/speedtest-tracker#493] [Q&A] [answered]
Hi all, 

Completely new here so be indulgent :-) - Thanks, 
I installed ST on a synology NAS within docker : the installation went without any problem : so far, nothing complicated. 

I have 2 issues :

1. i can not open the settings>general : i get a server error : "error 500" (the other sections work)
2. when running a test a get the following error {"type":"log","timestamp":"2023-03-15T21:00:15Z","message":"Configuration - Could not retrieve or read configuration (ConfigurationError)","level":"error"}

Any hint on those issues? 

I reinstalled the docker multiple times .... with always the same error. 

Thank you 

Stéphane

**Comment by alexjustesen** [comment]
Hi Stephane! It's likely a data issue that occurred. Try using a database IDE like [DBeaver](https://dbeaver.io/) to browse the app's tables. You can take one of the two steps below to correct the `settings` table.

1. Easier but destructive: run `php artisan app:install --force` in the container's terminal. This will COMPLETELY reset the DB so you will lose result history.
2. Using the screenshot below you can check your table values against the default and make corrections where needed. Once you update the database restart the container so the cache is reset.
<img width="569" alt="image" src="https://user-images.githubusercontent.com/1144087/225460964-ae94ba76-9fd9-44eb-ad41-ac76a28a96eb.png">

**Comment by stephdeville** [comment]
Hi Alex, 

Below is how that settings table look like. 
It seems that it is pretty much the same as the one you posted. 

<img width="743" alt="image" src="https://user-images.githubusercontent.com/6465135/225733625-b7f7e72c-5c11-4f80-a78d-51f888dd8645.png">

I will restart the whole thing ... the NAS, and try I clean install again and see. I find it weird that I m the only one having this issue so it has definitely something to do with my setup. 

Thank you

**Comment by stephdeville** [comment]
All rebooted, no improvement. 

Information about what i have running : 

Laravel v9.52.4
PHP v8.1.16
Filament v2.17.15
Speedtest Tracker v0.11.10

This is the only error i can see when i start the container :

❌ There seems to be a failure in checking the web server + PHP-FPM. Here's the response:

... and then the next line is empty.

**Comment by stephdeville** [comment]
Something interesting : 

I wanted to unminimize the container installation to have more tools and noted that it actually does not have any connection to the internet. 

Could be a DNS issue : IDK yet. 

I do have other containers running but the 3 other ones are running on the host network (no issue at all with those ones) and the only container running on the bridge network is speedtest-tracker.

**Reply by stephdeville** [reply]
Hi Alex, 

Found the root cause : when the firewall of my NAS is disabled : i do not have the issues mentioned here and everything works.. 
The last line of the fw rules I have is to deny every traffic i did not specifically allowed. 

See 👎 
<img width="798" alt="image" src="https://user-images.githubusercontent.com/6465135/225753585-280f37de-1e0f-462c-96ea-78f80ba8abd0.png">

If I have that last line deactivated ... it works as well. 

What I can not explain is that I actually have allowed in my rules the following ports: 

<img width="612" alt="image" src="https://user-images.githubusercontent.com/6465135/225754789-07be94a6-4332-4c88-95c3-a144c6417709.png">

Have you maybe any hint for me? Any additional rule i need to add? (I would like to have that last rule activated)

Thank you (and sorry for the spam today)

**Comment by stephdeville** [comment, marked as answer]
SOLVED : here are the last 2 lines in my fw: 

<img width="787" alt="image" src="https://user-images.githubusercontent.com/6465135/225764548-1e23d713-908a-4364-9d79-9196b56efc8d.png">

Thank you Alex :-)

**Reply by coumbsek** [reply]
Same issue here but disabling the firewall did not solve it. I am not sure either to understand the rule you add and why it should work. You state to the nas that any incoming connexion from it's own IPs (in bridge mode) are allowed to any port ? Did you keep the rule explicitly allowing for the 3 ports mapped to 80,443 and 9000 ?
